docs/topics/native/native-improving-compilation-time.md

```diff
@@ -0,0 +1,69 @@
+with Kotlin/Native targets.
+
+Read on for our tips on how to speed up the Kotlin/Native compilation process.
+
+## General recommendations
+
+* **Use the most recent version of Kotlin**. This way you will always have the latest performance improvements.
+* **Avoid creating huge classes**. They take a long time to compile and load during execution.
+* **Preserve downloaded and cached components between builds**. When compiling projects, Kotlin/Native downloads the required components
+  and caches some results of its work to the `$USER_HOME/.konan` directory. The compiler uses this directory for subsequent
+  compilations, making them take less time to complete.
+
+  When building in containers (such as Docker) or with continuous integration systems, the compiler may have to create
+  the `~/.konan` directory from scratch for each build. To avoid this step, configure your environment to preserve `~/.konan`
+  between builds. For example, redefine its location using the `KONAN_DATA_DIR` environment variable.
+
+## Gradle configuration
+
+The first compilation with Gradle usually takes more time than subsequent ones due to the need to download the dependencies,
+build caches, and perform additional steps. You should build your project at least twice to get an accurate reading of
+the actual compilation times.
+
+Here are some recommendations for configuring Gradle for better compilation performance:
+
+* **Increase the [Gradle heap size](https://docs.gradle.org/current/userguide/performance.html#adjust_the_daemons_heap_size)**.
+  Add `org.gradle.jvmargs=-Xmx3g` to `gradle.properties`. If you use [parallel builds](https://docs.gradle.org/current/userguide/performance.html#parallel_execution),
+  you might need to make the heap even larger or choose the right number of threads with `org.gradle.parallel.threads`.
+
+* **Build only the binaries you need**. Don't run Gradle tasks that build the whole project, such as `build` or `assemble`,
+  unless you really need to. These tasks build the same code multiple times, increasing the compilation time. In typical
+  cases such as running tests from IntelliJ IDEA or starting the app from Xcode, the Kotlin tooling avoids executing unnecessary
+  tasks. 
+  
+  If you have a non-typical case or build configuration, you might need to choose the task yourself.
+    * `linkDebug*`: To run your code during development, you usually need only one binary, so running the corresponding
+      `linkDebug*` task should be enough. Keep in mind that compiling a release binary (`linkRelease*`) takes more time than compiling a debug one
+    * `packForXcode`: Since iOS simulators and devices have different processor architectures, it's a common approach to
+      distribute a Kotlin Native binary as a universal (fat) framework. During local development, it will be faster to build
+      the `.framework` for only the platform you’re using. You can do this by calling the  `packForXcode` task generated
```

Comment:
In all other recommendations, we say imperatively – add, remove. Maybe we should rephrase this "You can" for consistency.